src/React.re

```diff
@@ -186,41 +186,59 @@ external useReducerWithMapState:
   ('state, 'action => unit) =
   "useReducer";
 
+/** A hook dependency. */
```

Comment:
To be clear: we don't actually need this `dep` type in the ReasonReact binding. All we need is the hooks that take dependencies to take an `array('a)` or `array(someType)`. The point is that it's an array of a fixed type, where the array would be empty e.g. `[||]`, contain a single element e.g. `[|somDep|]`, _or_ could contain elements wrapped in some unboxed GADT like `dep`. I just chose to make it `array(dep)` to provide an out-of-the-box solution and lock it in. If e.g. BuckleScript shipped with this unboxed GADT e.g. `[@unboxed] type any = Any(_): any`, we could use that.